plugin/rewrite/reverter.go

```diff
@@ -21,23 +22,69 @@ type ResponseRule struct {
 // dig will complain with ';; Question section mismatch: got example.org/HINFO/IN'
 type ResponseReverter struct {
 	dns.ResponseWriter
-	originalQuestion dns.Question
-	ResponseRewrite  bool
-	ResponseRules    []ResponseRule
+	request           *dns.Msg
+	originalQuestion  dns.Question
+	originalOPTHeader dns.RR_Header
+	ResponseRewrite   bool
+	ResponseRules     []ResponseRule
+}
+
+// align the OPT inside the msg the information provided - hasOPT and Header
+// the sub array of EDNS0 is not changed (unless OPT is removed)
+func replaceOPTHeader(m *dns.Msg, hasOpt bool, hdr dns.RR_Header) {
```

Comment:
boolean params are a good indicators this should be two functions

Reply:
I can change that.

Reply:
not two functions but integrate the boolean inside the function.